src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitExtension.java

```diff
@@ -230,22 +232,30 @@ public GetJUnitTestResults(Run<?, ?> build, HPRunnerType hpRunnerType, List<File
 
 
 			if (HPRunnerType.UFT.equals(hpRunnerType) || HPRunnerType.UFT_MBT.equals(hpRunnerType)) {
-				Node node = JenkinsUtils.getCurrentNode(workspace);
-				this.nodeName = node != null && !node.getNodeName().isEmpty() ? node.getNodeName() : "";
+				List<Node> nodes = new ArrayList<>();
+
+				allWorkspaces.forEach(workspace-> nodes.add(JenkinsUtils.getCurrentNode(workspace)));
+				nodes.forEach(node -> this.nodeNames.add(node != null && !node.getNodeName().isEmpty() ? node.getNodeName() : ""));
```

Comment:
It seems that there is case when node name is empty as far as I remember it a case when UFT is run on Jenkins machine itself and not on one of it's agents, can you please try to check if your code covers case the UFT test run with not agents?
Secondly instead of define array and fill it up there more convenient way to do it in Java like this
allWorkspaces.stream().filter(workspace-> nodes.add(JenkinsUtils.getCurrentNode(workspace)).map(workspace-> JenkinsUtils.getCurrentNode(workspace))).collect(collectors.toList())

Reply:
Indeed, when running on the Jenkins machine itself, the node name remains empty. I tested the behavior, and the reports URLs are created correctly in this case, with the path to the 'artifacts' folder from the Jenkins machine.

For the second part, I changed the code accordingly. Please let me know if I missed something.